src/signatures.cpp

```diff
@@ -349,10 +348,11 @@ g1 derive_child_g1_unhardened(
     sha.digest(digest.data());
 
     array<uint64_t, 4> nonce = scalar::fromBytesBE<4>(span<uint8_t, 32>(digest.begin(), digest.end()));
-    array<uint64_t, 4> quotient = {0, 0, 0, 0};
-    array<uint64_t, 4> remainder = {0, 0, 0, 0};
-    array<uint64_t, 4> q = fp::Q;
-    bn_divn_low(quotient.data(), remainder.data(), nonce.data(), 4, q.data(), 4);
+    array<uint64_t, 4> quotient = {0};
+    array<uint64_t, 4> remainder = {0};
+
+    bn_divn_safe(quotient, remainder, nonce, fp::Q);
+
     nonce = {remainder[0], remainder[1], remainder[2], remainder[3]};
```

Comment:
We don't need to copy `remainder` into `nonce`. Just return `pk.add(g1::one().mulScalar(remainder));`